File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/IceBreaker.java
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest;
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager;

/* JADX INFO: loaded from: classes3.dex */
public class IceBreaker extends Quest {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void calculateTargetProgress(int i) {
        this.targetProgress = ((long) (i - 5)) * 5;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    protected void configure() {
        this.idName = R.string.quest_ice_breaker_name;
        this.idDescription = R.string.quest_ice_breaker_description;
        this.defaultRarity = 1;
        this.minimumDifficulty = 6;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
    public void realignStaticReference() {
        QuestsManager.iceBreaker = this;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/quests/instances/IceBreaker.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.quests.instances

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.Quest
import it.paranoidsquirrels.idleguildmaster.storage.data.quests.QuestsManager

class IceBreaker : Quest() {
    override fun calculateTargetProgress(difficulty: Int) {
        targetProgress = ((difficulty - 5).toLong() * 5).toLong()
    }

    override fun configure() {
        idName = R.string.quest_ice_breaker_name
        idDescription = R.string.quest_ice_breaker_description
        defaultRarity = 1
        minimumDifficulty = 6
    }

    override fun realignStaticReference() {
        QuestsManager.iceBreaker = this
    }
}
